cmd/storage-class.go

```diff
@@ -115,68 +128,39 @@ func parseStorageClass(storageClassEnv string) (sc storageClass, err error) {
 	return sc, nil
 }
 
-// Validates the parity disks for Reduced Redundancy storage class
-func validateRRSParity(rrsParity, ssParity int) (err error) {
-	disks := len(globalEndpoints)
-	// disks < 4 means this is not a erasure coded setup and so storage class is not supported
-	if disks < 4 {
+// Validates the parity disks.
+func validateParity(ssParity, rrsParity int) (err error) {
+	if ssParity == 0 && rrsParity == 0 {
+		return nil
+	}
+
+	if !globalIsXL && !globalIsDistXL {
```

Comment:
`globalIsXL` is `true` for both single node and multi-node erasure setups. so, `if !globalIsXL {...}` is enough.

Reply:
the reason is to avoid any confusion in future, it is purely incidental that globalIsXL is true in case of distributed as well.

Reply:
Hmm, it doesn't look coincidental - See https://github.com/minio/minio/blob/a337ea4d11eb2ae87144a76cdefd6f77ae7303a4/cmd/server-main.go#L112
Other places in our code that rely on `globalIsXL` to be `true` in both single-node erasure backend and distributed setup are,
- https://github.com/minio/minio/blob/a337ea4d11eb2ae87144a76cdefd6f77ae7303a4/cmd/admin-rpc-server.go#L100
- https://github.com/minio/minio/blob/018813b98f4fb5e1f0572c820708bd6112e97b8f/cmd/admin-handlers.go#L463

So, it seems like a property that is here to remain.

Reply:
Okay then its fine.